synapse/types.py

```diff
@@ -916,3 +916,9 @@ class UserProfile(TypedDict):
     user_id: str
     display_name: Optional[str]
     avatar_url: Optional[str]
+
+
+@attr.s(auto_attribs=True, frozen=True)
```

Comment:
```suggestion
@attr.s(auto_attribs=True, frozen=True, slots=True)
```

Reply:
TIL

Reply:
My understanding is that we should pretty much use it everywhere. 👍